### PR TITLE
Fix script tag injection in MV3

### PR DESCRIPTION
**Pull request by jpuri**
#13972

Fix script tag injection in MV3.

### CODE REVIEW

File: app/scripts/background.js [hunk 1]
```diff
@@ -480,8 +480,8 @@ function setupController(initState, initLangCode) {
     if (count) {
       label = String(count);
     }
-    browser.browserAction.setBadgeText({ text: label });
-    browser.browserAction.setBadgeBackgroundColor({ color: '#037DD6' });
+    // browser.browserAction.setBadgeText({ text: label });
+    // browser.browserAction.setBadgeBackgroundColor({ color: '#037DD6' });
```
Comment (jpuri) [inline app/scripts/background.js:484]: Ignore above as it is different mv3 issue we plan to address in separate task.

File: app/scripts/contentscript.js [hunk 2]
```diff
@@ -37,12 +37,12 @@ if (shouldInjectProvider()) {
  *
  * @param {string} content - Code to be executed in the current document
  */
-function injectScript(content) {
+function injectScript() {
```
Comment (adonesky1) [inline app/scripts/contentscript.js:40]: Should `injectScript(inpageBundle);` become `injectScript();` on line 31?

Comment (adonesky1) [reply app/scripts/contentscript.js:40]: And remove `inpageSuffix` and `inpageBundle` consts?

Comment (jpuri) [reply app/scripts/contentscript.js:40]: I am thinking of adding a feature flag for MV3 and using it here to conditionally inject script text or source.

**Review by adonesky1 — DISMISSED** [review]

**Comment by jpuri** [comment]
Additionally script injection using src attribute is async and thus use of events like `DOMContentLoaded` can fail

https://github.com/MetaMask/test-dapp/blob/main/src/index.js#L1274

Using `window.onload` will be better option.

**Pull request merged**